spec/models/assignment_spec.rb

```diff
@@ -563,6 +636,71 @@
     end
   end
 
+context 'An Assignment' do
+  before :each do
+    @assignment = create(:assignment, group_name_autogenerated: false)
+  end
+
+  context 'as a noteable' do
+    it 'display for note without seeing an exception' do
+      assignment = create(:assignment)
+      if assignment.valid?
+        assignment.display_for_note
+      end
+    end
+  end
+
+  context 'with a student in a group with a marked submission' do
+    before :each do
+      @membership = create(:student_membership, grouping: create(:grouping, assignment: @assignment),
+                           membership_status: StudentMembership::STATUSES[:accepted])
+      sub = create(:submission, grouping: @membership.grouping)
+      @result = sub.get_latest_result
+
+      @sum = 0
+      [2,2.7,2.2,2].each do |weight|
+        create(:mark, {mark: 4, result: @result, markable: create(:rubric_criterion,
+                                                                  {assignment: @assignment, max_mark: weight * 4})})
+        @sum += weight
+      end
+      @total = @sum * 4
+    end
+
+    it 'return the correct maximum mark for rubric criteria' do
+      expect(@total).to eq (@assignment.max_mark)
+    end
+
+    it 'return the correct group for a given student' do
+      expect(@membership.grouping.group).to eq(@assignment.group_by(@membership.user))
+    end
+  end
+
+  context "with some groupings with students and ta's assigned " do
+    before :each do
+      (1..5).each do
+        grouping = create(:grouping, assignment: @assignment)
+        (1..3).each do
+          create(:student_membership, {grouping: grouping, membership_status: StudentMembership::STATUSES[:accepted]})
+        end
+        create(:ta_membership, {grouping: grouping, membership_status: StudentMembership::STATUSES[:accepted]})
+      end
+    end
+
+    it "be able to have it's groupings cloned correctly" do
+      clone = create(:assignment, {group_min: 1, group_max: 1})
+      number = StudentMembership.all.size + TaMembership.all.size
+      clone.clone_groupings_from(@assignment.id)
+
+      expect(@assignment.group_min).to eql(clone.group_min)
+      expect(@assignment.group_max).to eql(clone.group_max)
+      expect(@assignment.groupings.size).to eql(clone.groupings.size)
```

Comment:
The logic here is actually good. The problem is that `clone.groupings` needs to be "reloaded" to obtain the updated value (with the 5 groups created). You can do this by calling `clone.groupings.reload` above this line.